### PR TITLE
Feature/sprint121023/#862kft5p4 query to support

### PR DESCRIPTION
**Pull request by bulgarchik**

### CODE REVIEW

File: FastBank.Services/MessageService/MessageService.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                Console.WriteLine($"Subject: {message.Subject}");
+                Console.WriteLine($"Text: {message.Text}");
+                Console.WriteLine($"Status: {message.Status}");
+                Console.WriteLine(new string('*', Console.WindowWidth));
+                Console.WriteLine(new string(' ', Console.WindowWidth));
+            }
+            Console.ReadKey();
+        }
+
+        public void AddMessage(string subject, string text, MessageStatuses status, MessageType type, Customer sender, Customer? receiver, Roles receiverRole, Message basedOnMessage)
```
[the remark this answers is not in the document]
Comment (AngelovGrowth) [reply FastBank.Services/MessageService/MessageService.cs:36]: Customer sender and Customer receiver? 
This method definately should be revised. At least you need to think about the parameters and why you need to store each parameter in database.

File: FastBank.Infrastructure/DTOs/MessageDTO.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        public string Subject { get; private set; } = string.Empty;
+        public Guid? BasedOnMessageId { get; private set; }
+        [ForeignKey(nameof(BasedOnMessageId))]
+        public MessageDTO? BasedOnMessage { get; private set; }
+        public MessageStatuses Statuses { get; private set; }
+        public MessageType Type { get; private set; }
+
+        public Message? ToDomainObj()
+        {
+            return new Message(MessageId, Sender.ToDomainObj(), Receiver?.ToDomainObj(), ReceiverRole, Text, Subject, BasedOnMessage?.ToDomainObj(), Statuses, Type);
```
Comment (AngelovGrowth) [inline FastBank.Infrastructure/DTOs/MessageDTO.cs:46]: Wrap parameters

File: FastBank.Domain/Message.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+﻿namespace FastBank.Domain
+{
+    public class Message
+    {
+       
+        public Message(Guid messageId, Customer sender, Customer? receiver, Roles receiverRole, string text, string subject, Message? basedOnMessage, MessageStatuses messageStatuses, MessageType messageType)
```
Comment (AngelovGrowth) [inline FastBank.Domain/Message.cs:6]: Wrap parameters

File: FastBank.Domain/Message.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        public Message(Guid messageId,
+                       Customer sender,
+                       Customer? receiver,
+                       Roles receiverRole,
+                       string text,
+                       string subject,
+                       Message? basedOnMessage,
+                       MessageStatuses messageStatuses,
+                       MessageType messageType)
```
Comment (AngelovGrowth) [inline FastBank.Domain/Message.cs:13]: image
See my example of correctly wrapped method parameters

File: FastBank.Domain/MessageStatuses.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿namespace FastBank.Domain
+{
+    public enum MessageStatuses
+    {
+        Sent,
+        Delivered,
+        Accepted,
+        Replied
+    }
```
Comment (AngelovGrowth) [inline FastBank.Domain/MessageStatuses.cs:9]: Put this in Message.cs. It is related to the message class and it is good to be there.

File: FastBank.Domain/MessageType.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FastBank.Domain
+{
+    public enum MessageType
+    {
+        Inquery,
+    }
```
Comment (AngelovGrowth) [inline FastBank.Domain/MessageType.cs:12]: Put this in Message.cs. It is related to the message class and it is good to be there.

File: FastBank.Services/MessageService/MessageService.cs [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+            Console.WriteLine("Please text message");
+            var text = Console.ReadLine();
+
+            var message = new Message(Guid.NewGuid(), customer, null, Roles.CustomerService,
+                                      text, subject, null, MessageStatuses.Sent, MessageType.Inquery);
+            
+            AddMessage(message);
+            return message;
+        }
```
Comment (AngelovGrowth) [inline FastBank.Services/MessageService/MessageService.cs:63]: Please refactor. Use only one method for creating the message. In InputMessage you take the input from the user and then you create the message. Afterwards, you are going to AddMessage and again you create a new message. Why? Get the input, build the message and save it to the repo. That's it. Please, keep your code and follow the following priciple - KISS. Reference: https://dev.to/kwereutosu/the-k-i-s-s-principle-in-programming-1jfg

File: FastBank.Services/MessageService/MessageService.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (AngelovGrowth) [reply FastBank.Services/MessageService/MessageService.cs]: I'm OK to use inheritance. Let's do it as you suggest.

[the remark this answers is not in the document]
Comment (bulgarchik) [reply FastBank.Services/MessageService/MessageService.cs]: Can we have conversation about it?

**Review by AngelovGrowth — APPROVED** [review]

**Pull request merged**